secboot/secboot_sb.go

```diff
@@ -348,7 +348,7 @@ func AddBootstrapKeyOnExistingDisk(node string, newKey keys.EncryptionKey) error
 // Rename key slots on LUKS2 container. If the key slot does not
 // exist, it is ignored. If cryptsetup does not support renaming, then
 // the key slots are instead removed.
-func RenameOrDeleteKeys(node string, renames map[string]string) error {
+func RenameKeys(nonAtomic *NonAtomicOperationAllowedFlag, node string, renames map[string]string) error {
```

Comment:
this is a very weird API tbh. The call sites where this isn't passed will see a bare `nil` as the first argument which is even more confusing than bare true/false. I understand canonical/secboot does this, but it doesn't make it any less weird and it doesn't need to be exposed outside of the secboot package.

If this needs an argument then pass it in a struct:
```
type RenameKeysFlags struct {
    AllowNonAtomic bool
}

func RenameKeys(node string, renames map[string]string, flags RenameKeysFlags) error {
..
```

Reply:
It is on purpose weird because should avoid using this mode. Chris was concerned that would get used when we do not need to.

Reply:
As requested by @pedronis  this was simplified.